lib/frontman/commands/serve.rb

```diff
@@ -84,7 +85,35 @@ def serve
 end
 
 class FrontManServer < Sinatra::Base
-  set :port, 4568
+  port = default_port = Frontman::Config.get(:port, fallback: 4568)
+  num_retries = Frontman::Config.get(:port_retries, fallback: 1)
+
+  port_retry_strategy = Frontman::Config.get(:port_retry_strategy, ->(port_num){
+    (default_port + num_retries).times do
+      begin
+        port_in_use = Socket.tcp("localhost", port, connect_timeout: 3) { true }
+      rescue SocketError
+        port_in_use = false
+      end
+
+      if port_in_use
+        port_num += 1 
+      else
+        break
+      end
+    end
+
+    if port_in_use
+      raise Frontman::Error.new("Server failed to attach to a port")
+    end
+
```

Comment:
Instead of a generic `Frontman::Error`, could you please add a specific error that extends the generic `Frontman::Error` class for this scenario, e.g. `Frontman::ServerPortError` (you can decide on the name).  This way, we can define the message in the specific class, and we can change this check to something like:

```ruby
raise Frontman::ServerPortError if port_in_use
```

I also think we can mention the `:port_retries` config value in that error 🙂

Reply:
I have abstracted the error. I am sure the error message could be improved.